src/Ocelot/Configuration/Validator/FileConfigurationFluentValidator.cs

```diff
@@ -101,6 +109,24 @@ private static bool IsPlaceholderNotDuplicatedIn(string upstreamPathTemplate)
             var upstreamPathPlaceholders = matches.Select(m => m.Value);
             return upstreamPathPlaceholders.Count() == upstreamPathPlaceholders.Distinct().Count();
         }
+
+        private static bool SameNamePlaceholdersInUpstreamandDownstream(FileRoute route)
+        {
+            var regExPlaceholder = new Regex("{[^}]+}");
```

Comment:
I don't like this hacky expression!
I'm reading it as: "after opening curly bracket there is a set of chars which is not closing curly bracket, and finally ends with closing curly bracket". :rofl: 

Aly, have you ever heard about matching groups of `RegEx`?
- [Grouping constructs in regular expressions](https://learn.microsoft.com/en-us/dotnet/standard/base-types/grouping-constructs-in-regular-expressions)

I guess the expression like `({\w+})` should work. Then you have to read items from `match.Groups` property.

Reply:
I know regular expressions pretty well, I just copied the regex expression used in the function above thinking that since it exists in the code base it means it's ocelot's approved way of doing things, but I don't like it either, I will rewrite it in the correct way and I will fix the older one

Reply:
Good luck! :wink:

Reply:
@AlyHKafoury Your old regex worked, but I modified it to favor my suggestion of using `\{\w+\}`, no groups

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1927/commits/68ba69a222f9dfe4ef094803dd2348c9b19d97da ✔️